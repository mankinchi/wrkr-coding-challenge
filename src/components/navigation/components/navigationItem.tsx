import clsx from "clsx";
import { Link, useLocation } from "react-router-dom";

interface Props {
	title: string;
	path: string;
}

export const NavigationItem = (props: Props) => {
	const { title, path } = props;

	const { pathname } = useLocation();

	return (
		<Link to={path}>
			<div
				className={clsx(
					"cursor-pointer border-b border-slate-600 px-2 py-4 transition-all hover:font-bold",
					pathname === path && "font-semibold",
				)}
			>
				{title}
			</div>
		</Link>
	);
};
