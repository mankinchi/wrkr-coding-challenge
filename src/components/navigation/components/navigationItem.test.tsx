import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { NavigationItem } from "./navigationItem";

describe("Navigation Item", () => {
	it("should render correctly when not in path", () => {
		const { container } = render(
			<MemoryRouter initialEntries={["/"]}>
				<NavigationItem
					title="Test"
					path="/testPath"
				/>
			</MemoryRouter>,
		);

		expect(container).toMatchSnapshot();
	});

	it("should render correctly when in path", () => {
		const { container } = render(
			<MemoryRouter initialEntries={["/testPath"]}>
				<NavigationItem
					title="Test"
					path="/testPath"
				/>
			</MemoryRouter>,
		);

		expect(container).toMatchSnapshot();
	});
});
