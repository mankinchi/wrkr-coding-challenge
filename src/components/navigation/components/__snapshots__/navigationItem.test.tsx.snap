// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Item should render correctly when in path 1`] = `
<div>
  <a
    href="/testPath"
  >
    <div
      class="cursor-pointer border-b border-slate-600 px-2 py-4 transition-all hover:font-bold font-semibold"
    >
      Test
    </div>
  </a>
</div>
`;

exports[`Navigation Item should render correctly when not in path 1`] = `
<div>
  <a
    href="/testPath"
  >
    <div
      class="cursor-pointer border-b border-slate-600 px-2 py-4 transition-all hover:font-bold"
    >
      Test
    </div>
  </a>
</div>
`;
